cypress/e2e/gui/testsGui.cy.js

```diff
@@ -0,0 +1,119 @@
+describe('Validate the GUI', () => {
+  beforeEach(() => {
+    const BASE_URL = `${Cypress.env('BASE_URL')}`
+    const CUSTOMERS_API_URL = `${Cypress.env('API_URL')}/customers`
+    cy.intercept({
+      method: 'GET',
+      url: `${CUSTOMERS_API_URL}?**`,
+    }).as('getCustomers');
+    cy.visitPageAndAcceptCookies()
+    cy.visit(BASE_URL)
+  })
+
+  it('Apply a filter, go to a customer page, return to the list page, and check if the previous filters are selected', () => {
+    // Filter the companies
+    cy.get('[data-testid="size-filter"]').select('Medium')
+    cy.get('[data-testid="industry-filter"]').select('Retail')
+    cy.wait('@getCustomers')
+    cy.get('[aria-label="View company: Kilback Co"]').click()
+    cy.contains("Company ID").should('be.visible')
+    // Back to companies list
+    cy.contains("Back").click()
+
+    cy.get('[data-testid="size-filter"]').should('have.value', 'Medium')
+    cy.get('[data-testid="industry-filter"]').should('have.value', 'Retail')
+  })
+
+  it('Move to custmer page and return to clients list', () => {
+    // Open the details page
+    cy.get('button[aria-label^="View company:"]').first().click()
```

Comment:
Veja uma solução mais simples e com uma cadeia de comandos a menos.

```suggestion
    cy.contains('button', 'View').click()
```

Além disso, o código fica mais curto.